onmt/trainer.py

```diff
@@ -307,8 +340,20 @@ def validate(self, valid_iter, moving_average=None):
 
         with torch.no_grad():
             stats = onmt.utils.Statistics()
-
+            i = 0
+            computed_stats = {}
             for batch in valid_iter:
+                if i == 0:
```

Comment:
That's not very pythonic. Please use `enumerate` if you need indices in the loop.

Reply:
Also, do we really need to do this in the loop? The `batch.dataset` is a hack anyway so it'd probably be "less hacky" if we did this outside of the loop, e.g. afterwards, on the last `batch`.